include/xrpl/beast/hash/xxhasher.h

```diff
@@ -24,29 +24,110 @@
 
 #include <xxhash.h>
 
+#include <array>
 #include <cstddef>
-#include <new>
-#include <type_traits>
+#include <cstdint>
+#include <optional>
+#include <span>
 
 namespace beast {
 
 class xxhasher
 {
+public:
+    using HashType = std::size_t;
```

Comment:
Please remove `HashType` and use `result_type` instead. Which already is defined (below), you just need to move it up here.